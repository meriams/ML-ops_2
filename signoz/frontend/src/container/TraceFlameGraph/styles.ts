import styled from 'styled-components';

const SPAN_HEIGHT = 10;
const SPAN_V_PADDING = 1;
export const TOTAL_SPAN_HEIGHT = SPAN_HEIGHT + 2 * SPAN_V_PADDING;

/**
 * An individual span for traces flame graph
 */
export const SpanItemContainer = styled.div<{
	topOffset: number;
	leftOffset: number;
	width: number;
	spanColor: string;
	selected: boolean;
	zIdx: number;
}>`
	position: absolute;
	top: ${(props): string | number => props.topOffset}px;
	left: ${(props): string | number => props.leftOffset}%;
	width: ${(props): string | number => props.width}%;
	height: ${SPAN_HEIGHT}px;
	margin: ${SPAN_V_PADDING}px 0;
	background-color: ${({ spanColor }): string | number => spanColor};
	border-radius: ${SPAN_HEIGHT / 2}px;
	z-index: ${(props): string | number => props.zIdx};
`;

/**
 * Container for spans, for traces flame graph.
 */
export const TraceFlameGraphContainer = styled.div<{
	height: number;
}>`
	position: relative;
	width: 100%;
	height: ${({ height }): string | number => height || 120}px;
`;
