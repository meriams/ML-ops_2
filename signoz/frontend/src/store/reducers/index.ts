import { combineReducers } from 'redux';

import appReducer from './app';
import globalTimeReducer from './global';
import { LogsReducer } from './logs';
import metricsReducers from './metric';
import { ServiceMapReducer } from './serviceMap';
import traceReducer from './trace';
import { usageDataReducer } from './usage';

const reducers = combineReducers({
	traces: traceReducer,
	usageDate: usageDataReducer,
	globalTime: globalTimeReducer,
	serviceMap: ServiceMapReducer,
	app: appReducer,
	metrics: metricsReducers,
	logs: LogsReducer,
});

export type AppState = ReturnType<typeof reducers>;

export default reducers;
