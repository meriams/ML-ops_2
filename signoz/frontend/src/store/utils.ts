import getLocalStorageKey from 'api/browser/localstorage/get';
import { LOCALSTORAGE } from 'constants/localStorage';
import { User } from 'types/reducer/app';

export const getInitialUserTokenRefreshToken = (): AuthTokenProps | null => {
	const accessJwt = getLocalStorageKey(LOCALSTORAGE.AUTH_TOKEN);
	const refreshJwt = getLocalStorageKey(LOCALSTORAGE.REFRESH_AUTH_TOKEN);

	if (accessJwt && refreshJwt) {
		return {
			accessJwt,
			refreshJwt,
		};
	}

	return null;
};

interface AuthTokenProps {
	accessJwt: User['accessJwt'];
	refreshJwt: User['refreshJwt'];
}
