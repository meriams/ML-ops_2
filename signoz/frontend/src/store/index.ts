import {
	applyMiddleware,
	compose,
	legacy_createStore as createStore,
} from 'redux';
import thunk, { ThunkMiddleware } from 'redux-thunk';
import AppActions from 'types/actions';

import reducers, { AppState } from './reducers';

const composeEnhancers =
	(window && window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__) || compose;

const store = createStore(
	reducers,
	composeEnhancers(
		applyMiddleware(thunk as ThunkMiddleware<AppState, AppActions>),
	),
);

export type AppDispatch = typeof store.dispatch;

if (window !== undefined) {
	window.store = store;
}

export default store;
