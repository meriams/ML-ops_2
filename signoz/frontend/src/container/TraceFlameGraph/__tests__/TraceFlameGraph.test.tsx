import { render, renderHook } from '@testing-library/react';
import TraceFlameGraph from 'container/TraceFlameGraph';
import { useState } from 'react';
import { Provider } from 'react-redux';
import store from 'store';

test('loads and displays greeting', () => {
	const { rerender } = renderHook(() => useState(''));

	const { asFragment } = render(
		<Provider store={store}>
			<TraceFlameGraph
				{...{
					hoveredSpanId: '',
					intervalUnit: { multiplier: 0, name: 'm' },
					onSpanHover: rerender,
					onSpanSelect: (): void => {},
					selectedSpanId: '',
					traceMetaData: {
						globalEnd: 0,
						globalStart: 0,
						levels: 0,
						spread: 0,
						totalSpans: 0,
					},
					missingSpanTree: false,
					treeData: {
						children: [],
						id: '',
						name: '',
						serviceColour: '#a0e',
						serviceName: '',
						startTime: 0,
						tags: [],
						time: 100,
						value: 100,
						event: [],
						hasError: false,
						parent: undefined,
					},
				}}
			/>
		</Provider>,
	);
	expect(asFragment()).toMatchSnapshot();
});
