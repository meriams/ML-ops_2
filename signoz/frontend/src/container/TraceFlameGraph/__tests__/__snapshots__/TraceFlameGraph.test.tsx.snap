// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`loads and displays greeting 1`] = `
<DocumentFragment>
  .c1 {
  position: absolute;
  top: 0px;
  left: NaN%;
  width: Infinity%;
  height: 10px;
  margin: 1px 0;
  background-color: hsl(282.9,100%,60.7%);
  border-radius: 5px;
  z-index: 1;
}

.c0 {
  position: relative;
  width: 100%;
  height: 120px;
}

<div
    class="c0"
    height="0"
  >
    <div
      class="c1"
      title=""
      width="Infinity"
    />
  </div>
</DocumentFragment>
`;
